mesonbuild/cmake/interpreter.py

```diff
@@ -67,6 +67,14 @@
     'CMP0102',
 ]
 
+# CMAKE is a bit more averse to debugging, but in spirit the build types match
+buildtype_map = {
```

Comment:
```suggestion
BUILDTYPE_MAP: T.Mapping[str, str] = {
```

I know we're not doing this in the file right now, but let's keep with PEP8 and use all caps for module constants, and adding the `Mapping` annotation so mypy will complain if someone tries to mutate this

Reply:
Are you sure? I think it looks very weird next to for instance `target_type_map = {` just a few lines down. I'd rather leave this change for a follow-up pull requests that is more comprehensive.

Reply:
Yes. I'd prefer if new things were done correctly, even if we don't fix all of the old things at once. Otherwise the work of fixing it becomes harder and more time consuming, and thus less likely to be done at all

Reply:
I guess we can do this as a follow up.

Reply:
Haha, it was high on my list of TODOs for the next time I could spend time on open source projects. I may have a follow-up, including a release snippet, ready this weekend.

Reply:
I decided to go ahead and merge it since we have another PR that would benefit from leveraging this, so I went ahead and merged it. A snippet would be good though.